### PR TITLE
Convert to HPI

### PR DESCRIPTION
**Pull request by kzantow**
**Description**
- See [JENKINS-42602](https://issues.jenkins-ci.org/browse/JENKINS-42602).

NOTE: probably won't merge this until merging JDL into blueocean-plugin

**Submitter checklist**
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.jenkins-ci.plugins</groupId>
+		<artifactId>plugin</artifactId>
+		<version>2.26</version>
```
Comment (jglick) [inline pom.xml]: add `<relativePath />`

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+				</includes>
+			</resource>
+		</resources>
+		<plugins>
+			<!-- Disable InjectedTest -->
+			<plugin>
+				<groupId>org.jenkins-ci.tools</groupId>
+				<artifactId>maven-hpi-plugin</artifactId>
+				<configuration>
+					<disabledTestInjection>true</disabledTestInjection>
```
Comment (jglick) [inline pom.xml]: Do you wear a seat belt?

Comment (i386) [inline pom.xml]: I suspect this is because it is not a java/jelly plugin. Injected tests would do nothing?

File: pom.xml (hunk 1 above)
Comment (kzantow) [inline pom.xml]: Why is this?

**Comment by i386** [comment]
@kzantow btw we will have to open up a HOSTING ticket and add repository permissions so we can release this.

**Comment by kzantow** [comment]
@i386 yeah, this and core-js :)

**Comment by michaelneale** [comment]
if this is a submodule of blue ocean don't need a HOSTING ticket

**Comment by michaelneale** [comment]
I guess this will actually be going into main blueocean repo?

File: pom.xml (hunk 2 above)
Comment (jglick) [inline pom.xml]: Not much, but at least checks any plugin dependencies and that sort of thing.

**Pull request closed without merging**